force_bdss/io/workflow_writer.py

```diff
@@ -22,72 +22,14 @@ def write(self, workflow, path, *, mode="w"):
         mode: string
             file open mode, "w" by default to write
         """
-        data = {"version": self.version}
-        data["workflow"] = self._workflow_data(workflow)
+        data = {"version": self.version, "workflow": workflow.__getstate__()}
 
         with open(path, mode) as f:
             json.dump(data, f, indent=4)
 
     def get_workflow_data(self, workflow):
         """ Public method to get a serialized form of workflow"""
-        return self._workflow_data(workflow)
-
-    def _workflow_data(self, workflow):
-        workflow_data = {
-            "mco_model": self._mco_data(workflow.mco_model),
-            "execution_layers": [
-                self._execution_layer_data(el)
-                for el in workflow.execution_layers
-            ],
-            "notification_listeners": [
-                self._model_data(nl) for nl in workflow.notification_listeners
-            ],
-        }
-
-        return workflow_data
-
-    def _mco_data(self, mco):
-        """Extracts the data from the MCO object and returns its dictionary.
-        If the MCO is None, returns None"""
-        if mco is None:
-            return None
-
-        data = self._model_data(mco)
-
-        parameters_data = []
-        for param in data["model_data"]["parameters"]:
-            state = param.__getstate__()
-
-            parameters_data.append(
-                {"id": param.factory.id, "model_data": state}
-            )
-
-        data["model_data"]["parameters"] = parameters_data
-
-        kpis_data = []
-        for kpi in data["model_data"]["kpis"]:
-            kpis_data.append(kpi.__getstate__())
-
-        data["model_data"]["kpis"] = kpis_data
-
-        return data
-
-    def _execution_layer_data(self, layer):
-        """Extracts the execution layer list of DataSource models"""
-        data = []
-
-        for ds in layer.data_sources:
-            data.append(self._model_data(ds))
-
-        return data
-
-    def _model_data(self, model):
-        """
-        Extracts the data from an external model and returns its dictionary
-        """
-        state = model.__getstate__()
-
-        return {"id": model.factory.id, "model_data": state}
+        return workflow.__getstate__()
 
 
 def pop_recursive(dictionary, remove_key):
```

Comment:
It feels like `pop_recursive`, `pop_dunder_recursive` and `nested_getstate` could be moved to the same module as `BaseModel` since they are now used at a more fundamental level

Reply:
Sounds good. I need to have a look at that, since it might be that too many plugins import these functions from here. I will create an issue for this instead, if you don't mind.

Reply:
Done.